test/sjcl-ecdsa-recoverablepublickey-test.js

```diff
@@ -43,7 +43,7 @@ describe('ECDSA signing with recoverable public key', function(){
 
         var pub_val_point = secret_key._curve.G.mult(secret_key._exponent);
         var public_key = new sjcl.ecc.ecdsa.publicKey(curve, pub_val_point);
-        var hash = hash_function(message);
+        var hash = sjcl.bitArray.clamp(hash_function(message), secret_key._curveBitLength);
```

Comment:
> blobvault messaging code uses sha512.
Ah, I glossed over that fact. 

The custom valid-ecc sign/verify code didn't clamp the hash to qlen, where as the sjcl implementations do. Old signatures created before would no longer verify with this change . Hrmmm.

> lives in ripple-lib 
You're proposing extracting it into a separate module?